framework/src/utils/InputParameters.C

```diff
@@ -967,3 +969,19 @@ InputParameters::getParamHelper<MultiMooseEnum>(const std::string & name,
 {
   return pars.get<MultiMooseEnum>(name);
 }
+
+void
+InputParameters::setReservedValues(const std::string & name, const std::set<std::string> & reserved)
+{
+  _reserved_values.insert(std::make_pair(name, reserved));
+}
+
+std::set<std::string>
+InputParameters::reservedValues(const std::string & name) const
+{
+  auto it = _reserved_values.find(name);
+  if (it == _reserved_values.end())
+    return std::set<std::string>();
```

Comment:
Ah! This is why you are returning a copy. Although this isn't a performance critical part of the code, what we do in other similar situations is create a const private "empty" member that we can return for these. I'm willing to go either way here since this isn't a hotspot. If you choose to keep the copy approach, please document it in the declaration as to why you are purposely doing a container copy.